flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/typeutils/ExternalSerializer.java

```diff
@@ -59,15 +59,19 @@
 
     private final TypeSerializer<I> internalSerializer;
 
+    private final boolean bidirectional;
```

Comment:
I went with a simple flag to start with as a prototype. But we should discuss how we want to model this, boolean flags don't make for nice APIs.